### DISCUSSION TITLE
Touchscreen controls issue workaround

### DISCUSSION THREAD

**Discussion by Medard22** [Medard22/Dolphin-MMJR2-VBI#708] [Announcements]
Before Dolphin rewrote the input handling, it relied on two systems: the standard system used on PC and a SharedPreferences system for overlay controls to determine which controls to display. These systems can become unsynchronized, which was one of the reasons the Dolphin team overhauled input handling. MMJR2 still uses the old system, so you may encounter this issue yourself.

**This problem affects only touchscreen input. Physical controllers are unaffected, and you can bind controls or create per-game configurations as usual.**

- MMJR2 starts with Wiimote + Nunchuk as the default option.
- To modify touchscreen input, go to Settings -> Overlay Controls -> Choose Controller. This replaces the default configuration in the core emulator and displays the appropriate touchscreen controls.
- The touchscreen overlay settings are not saved per game. For example, if you play one game with a Horizontal Wiimote setup and then start another requiring Wiimote + Nunchuk, the overlay will remain set to Horizontal Wiimote. You will need to manually select the correct controller option.
- Similarly, if you start a game like Kirby and switch to Horizontal Wiimote, exit the game, and then restart it, the touchscreen will show the Horizontal Wiimote overlay, but the game will complain that a Nunchuk is connected.
- If you don't want to manually select controls every time in these games, you can create a custom per-game Wiimote setting.
- Long pres the game cover, select Wii Input, choose the correct extension (Classic Controller, or no extension), then return to the main menu. This will generate a per-game .ini file in mmjr2-vbi/Config/Profiles/Wiimote. The emulator will automatically load this file and apply the correct configuration.
- If you create a custom .ini file and want a  Horizontal Wiimote setup, make sure to change the D-Pad axes as follows.

D-Pad/Up = `Button 109`
D-Pad/Down = `Button 110`
D-Pad/Left = `Button 108`
D-Pad/Right = `Button 107`

I hope this makes sense.

**Comment by Omegaprime134** [comment]
Hello buddy uh....I have a problem with the emulator: The problem is that when I exit a wii game and then go back the IR Point stops to working when I try to play a Wii Game so I dont know if this normal or not.

**Comment by alifm58131-pixel** [comment]
Dang man. I was making the best touch overlay for skyward sword. But when i play re4, the overlay from skyward sword is applied to all games? How can i make new profile like this.
